src/components/relatedProducts/ComparisonModal.jsx

```diff
@@ -1,27 +1,51 @@
-import React, {useState, useEffect} from 'react'
+import React, { useState, useEffect } from 'react'
 import ReactDom from 'react-dom'
 import { useSelector, useDispatch } from 'react-redux';
-import {closeModal} from '../../redux/actions/index.js'
+import { closeModal } from '../../redux/actions/index.js'
 import api from '../../apis/RPandOL.js'
+import StarRatings from 'react-star-ratings'
 
-export default function ComparisonModal({onClose, id}) {
+export default function ComparisonModal({ onClose, id }) {
   const dispatch = useDispatch();
   const productId = useSelector((state) => state.productId);
-  const comparisonId = useSelector(state=>state.comparisonId);
+  const comparisonId = useSelector(state => state.comparisonId);
   const [baseDetails, setBaseDetails] = useState({});
   const [comparedDetails, setComparedDetails] = useState({});
-  let features = new Set([]);
+  const [baseStars, setBaseStars] = useState(0);
+  const [comparedStars, setComparedStars] = useState(0);
+  const [features, setFeatures] = useState([])
```

Comment:
just confirming - features is the combination (set) of features from both products you are comparing?

Reply:
Yep! 👍👍